ddtrace/_trace/span.py

```diff
@@ -631,11 +634,28 @@ def set_link(
             )
         )
 
-    def _set_link_object(self, link: SpanLink) -> None:
-        # We will be changing this behavior to allow certain kinds of span
-        # links to coexist in the _links list even if they have the same
-        # span_id. For now, we are basically reimplementing the old
-        # dictionary-like behavior.
+    def _add_span_pointer(
+        self,
+        pointer_kind: str,
+        pointer_direction: _SpanPointerDirection,
+        pointer_hash: str,
+        extra_attributes: Optional[Dict[str, Any]] = None,
+    ) -> None:
+        # This is a Private API for now.
+
+        self._set_link_or_append_pointer(
```

Comment:
I am not sure if this has been discussed but can we avoid storing span pointers and "regular" span links in the same  span property? This will allow us to only serialize span pointers when sending trace payloads to severless trace endpoints (in the jsonencoder). I don't think we should send span pointers to the Datadog Agent (at least in the near term).

We can also achieve this functionality by removing span pointers from Span._links in the MsgPackEncoders (in _encoding.pyx). Either works for me.

Reply:
Since this API is private we don't need to address this issue in this PR.

Reply:
Awesome! I just wanted to double check that trace intake was aware of this change. If they can support span links with trace_id and span_id equal to zero then that's perfect. We can keep this change as is.